File: java-test/src/main/java/post/parthmistry/nst/util/TimeoutChecker.java
package post.parthmistry.nst.util;

import java.time.Duration;
import java.time.LocalDateTime;

public class TimeoutChecker {

    private final LocalDateTime timeoutDateTime;

    public TimeoutChecker(Duration duration) {
        this.timeoutDateTime = LocalDateTime.now().plus(duration);
    }

    public boolean isTimeout() {
        return LocalDateTime.now().isAfter(this.timeoutDateTime);
    }

}

File: rust-test/src/timeout_checker.rs
use std::time::{Duration, Instant};

pub struct TimeoutChecker {
    timeout_instant: Instant
}

impl TimeoutChecker {

    pub fn new(duration: Duration) -> TimeoutChecker {
        TimeoutChecker {
            timeout_instant: Instant::now() + duration
        }
    }

    pub fn is_timeout(&self) -> bool {
        Instant::now() > self.timeout_instant
    }

}
